client/rpcflags/rpcflags_test.go

```diff
@@ -0,0 +1,134 @@
+	// requests against the Trillian services created above.
+	conn, err := grpc.Dial(logEnv.Address, dialOpts...)
+	if err != nil {
+		t.Errorf("failed to dial %v: %v", logEnv.Address, err)
+	}
+	defer conn.Close()
+
+	adminClient := trillian.NewTrillianAdminClient(conn)
+	_, err = adminClient.ListTrees(context.Background(), &trillian.ListTreesRequest{})
+	if err != nil {
+		t.Errorf("failed to request trees from the Admin Server: %v", err)
+	}
+}
+
+func TestNewClientDialOptionsFromFlagsWithTLSCertFileMissing(t *testing.T) {
+	defer flagsaver.Save().Restore()
+	err := flag.Set("tls_cert_file", "/a/missing/file")
+	if err != nil {
+		t.Errorf("Failed to set flag: %v", err)
+	}
+
+	dialOpts, err := NewClientDialOptionsFromFlags()
+	if err == nil || !strings.Contains(err.Error(), "no such file or directory") {
+		t.Errorf("Expected to get an error due to the file not being found, instead got: %v", err)
+	}
+
+	if dialOpts != nil {
+		t.Errorf("Expected returned dialOpts to be nil, instead got: %v", dialOpts)
+	}
+}
+
+func TestNewClientDialOptionsFromFlagsWithTLSCertFileSet(t *testing.T) {
+	// Create new TLS certificates for the test services, and write the Client
+	// certificate to a file (so we can refer to it using the flag).
+	crtFile, cleanupCrtFile := setup.TempFile(t, "test.crt.")
+	defer cleanupCrtFile()
+
+	tlsCert := setup.NewTLSCertificate(t)
+
+	// Certificate file and client dial options.
+	err := pem.Encode(crtFile, &pem.Block{Type: "CERTIFICATE", Bytes: tlsCert.Certificate[0]})
+	if err != nil {
+		t.Fatalf("Failed to encode the test TLS certificate %v", err)
+	}
+	clientCreds, err := credentials.NewClientTLSFromFile(crtFile.Name(), "")
+	if err != nil {
+		t.Fatalf("Failed to get credentials: %v", err)
+	}
+
+	// Set up Trillian servers (with TLS enabled)
+	const numSequencers = 2
+	serverCreds := credentials.NewServerTLSFromCert(&tlsCert)
+	serverOpts := []grpc.ServerOption{grpc.Creds(serverCreds)}
+	clientOpts := []grpc.DialOption{grpc.WithTransportCredentials(clientCreds)}
+	logEnv, err := integration.NewLogEnvWithGRPCOptions(context.Background(), numSequencers, serverOpts, clientOpts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer logEnv.Close()
+
+	// Set up the flag.
+	defer flagsaver.Save().Restore()
+	err = flag.Set("tls_cert_file", crtFile.Name())
+	if err != nil {
+		t.Errorf("Failed to set -tls_cert_file flag: %v", err)
+	}
+
+	dialOpts, err := NewClientDialOptionsFromFlags()
+	if err != nil {
+		t.Errorf("Got an unexpected error: %v", err)
+	}
+
+	// Check that the returned dial options can be used to connect and make
+	// requests against the Trillian services created above.
+	conn, err := grpc.Dial(logEnv.Address, dialOpts...)
+	if err != nil {
+		t.Errorf("failed to dial %v: %v", logEnv.Address, err)
+	}
+	defer conn.Close()
+
+	adminClient := trillian.NewTrillianAdminClient(conn)
+	_, err = adminClient.ListTrees(context.Background(), &trillian.ListTreesRequest{})
```

Comment:
can do the inline thing here too, since you're discarding the return value.

Reply:
Done.